### PR TITLE
More ref guide improvements

### PR DESCRIPTION
**Pull request by bryevdv**
This PR:

* cleans up `conf.py`
* adds support for `autoclasstoc` (so far just to models)

  Screen Shot 2021-04-07 at 3 48 58 PM

### CODE REVIEW

**Review by tcmetzger — COMMENTED** [review]
I built the docs locally and was very happy to see the TOC. This looks great! 
I'll add some replacements for rest_epilog shortly.

File: sphinx/source/conf.py [hunk 1]
```diff
@@ -53,26 +61,17 @@
...
-.. |Property|           replace:: :py:class:`~bokeh.core.property.bases.Property`
-.. |PropertyDescriptor| replace:: :py:class:`~bokeh.core.property.descriptor.PropertyDescriptor`
-.. |PropertyContainer|  replace:: :py:class:`~bokeh.core.property.wrappers.PropertyContainer`
-.. |PropertyUnitsSpec|  replace:: :py:class:`~bokeh.core.properties.PropertyUnitsSpec`
-.. |field|              replace:: :py:func:`~bokeh.core.properties.field`
-.. |value|              replace:: :py:func:`~bokeh.core.properties.value`
-"""
+needs_sphinx = '3.5'
+
+rst_epilog = open("rst_epilogue.txt").read()
```
Comment (tcmetzger) [inline sphinx/source/conf.py:66]: For consistency's sake, how about we rename the file to `rst_epilog.txt` to match the variable name `rst_epilog`?

Comment (bryevdv) [reply sphinx/source/conf.py:66]: I believe Sphinx will try to to parse it as a page in the docs if we do that, at least unless we move it up out of the `source` tree

Comment (bryevdv) [reply sphinx/source/conf.py:66]: sure that seems good

File: sphinx/source/docs/first_steps/first_steps_4.rst [hunk 2]
```diff
@@ -58,8 +58,7 @@ Setting width and height
 
 To set the size of your plot, use the attributes
 :class:`~bokeh.models.plots.Plot.plot_height` and
-:class:`~bokeh.models.plots.Plot.plot_width` when calling the
-:func:`~bokeh.plotting.figure` function:
+:class:`~bokeh.models.plots.Plot.plot_width` when calling the |figure| function:
```
Comment (bryevdv) [inline sphinx/source/docs/first_steps/first_steps_4.rst:61]: I think this should `width` and `height` now

File: sphinx/source/rst_epilogue.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.. |bokeh.models|       replace:: :ref:`bokeh.plotting <bokeh.plotting>`
+.. |bokeh.plotting|     replace:: :ref:`bokeh.plotting <bokeh.plotting>`
+.. |Color|              replace:: :py:class:`~bokeh.core.properties.Color`
+.. |DataSpec|           replace:: :py:class:`~bokeh.core.properties.DataSpec`
+.. |Document|           replace:: :py:class:`~bokeh.document.Document`
+.. |field|              replace:: :py:func:`~bokeh.core.properties.field`
+.. |figure|             replace:: py:func:`~bokeh.plotting.figure`
```
Comment (bryevdv) [inline sphinx/source/rst_epilogue.txt:7]:
```suggestion
.. |figure|             replace:: :py:func:`~bokeh.plotting.figure`
```

File: sphinx/source/docs/first_steps/first_steps_4.rst (hunk 2 above)
Comment (tcmetzger) [reply sphinx/source/docs/first_steps/first_steps_4.rst:61]: Good point!

File: sphinx/source/rst_epilogue.txt (hunk 3 above)
Comment (tcmetzger) [reply sphinx/source/rst_epilogue.txt:7]: Did copy and paste a bit too quickly here 😊

**Comment by tcmetzger** [comment]
BTW: Changing `plot_width` and `plot_height` to `width` and `height` means this pr also fixes #10980!

**Pull request merged**